typing/datarepr.ml

```diff
@@ -116,7 +111,7 @@ let constructor_descrs ~current_unit ty_path decl cstrs =
         in
         let (tag, descr_rem) =
           match cd_args with
-          | _ when decl.type_unboxed.unboxed ->
+          | _ when rep = Variant_unboxed ->
```

Comment:
I would prefer `match rep, cd_args`, which would avoid the implicit fragile pattern matching here.

Reply:
Done

Reply:
(I still see `when rep = ..`, did you forgot to update your remote branch?)